Explorer/Assets/DCL/AvatarAnimation/Editor/TriggerEmotePlayableBehaviourcs.cs

```diff
@@ -0,0 +1,96 @@
+﻿
+using Arch.Core;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.AvatarRendering.Emotes;
+using DCL.Profiles;
+using Global.Dynamic;
+using UnityEngine.Playables;
+
+namespace DCL.AvatarAnimation
+{
+    /// <summary>
+    /// A playable / clip for the Unity timeline that takes a URN (either local or remote) and tells an avatar to play it.
+    /// </summary>
+    public class TriggerEmotePlayableBehaviour : PlayableBehaviour
+    {
+        public string URN;
+
+        private Entity cachedEntity = Entity.Null;
+        private AvatarBase cachedAvatar;
+
+        public override void ProcessFrame(Playable playable, FrameData info, object playerData)
+        {
+            if(string.IsNullOrEmpty(URN))
+                return;
+
+            // Gets the asset associated to the track
+            AvatarBase avatar = (AvatarBase)playerData;
+
+            // Prevents the emote from playing while there is an emote already playing
+            if(cachedEntity != Entity.Null && GlobalWorld.ECSWorldInstance.TryGetRef<CharacterEmoteComponent>(cachedEntity, out bool emoteExists).CurrentEmoteReference != null)
```

Comment:
You should check if `TryGetRef` returns `true` before accessing its fields. Otherwise you are at risk of `NullReferenceException` even for value types as you retrieve it by `ref`